unittests/array_bug_tester.py

```diff
@@ -55,6 +55,65 @@ def test4(self):
             '::xyz[2][3]' == aaaa_type.decl_string,
             aaaa_type.decl_string)
 
+    def test5(self):
+        code = 'char const arr[4] = {};'
+        src_reader = parser.source_reader_t(self.config)
+        global_ns = declarations.get_global_namespace(
+            src_reader.read_string(code))
+        arr_type = global_ns.variable('arr').type
+        if self.config.xml_generator == "gccxml":
+            self.assertTrue(
+                'char[4] const' == arr_type.decl_string,
```

Comment:
No it doesn't make sense. I guess this is a gccxml bug. From what I know, the gccxml project did a lot of work to produce the xml file, and the output was not always very "clean". This may also be due to how it interacted with gcc.

With castxml, we rely on the AST produced by clang/llvm. And this AST is much more robust, as far I have seen. The ordering in the xml file is always correct (there were other cases were the output was nicer when I added support for CastXML).

So just leave the tests like this. It just makes sure the results stay the same in the future. CastXML will be the default xml parser in release 1.8.0, and gccxml support will be dropped in further releases, so these gccxml specific checks will be dropped anyway.